tests/unittest/inputs/test_mixed_modal_encode_context.py

```
@@ -0,0 +1,219 @@
+    assert ctx.embedding_lengths == (32, 48, 32)
+    # prompt_pos of item i is just its index in `order`.
+    assert ctx.order[2] == ("image", 1)
+
+
+def test_length_mismatch_raises():
+    with pytest.raises(ValueError, match="embedding_lengths"):
+        MixedModalItemOrder(order=(("image", 0),), embedding_lengths=(32, 48))
+
+
+def test_duplicate_reference_raises():
+    with pytest.raises(ValueError, match="more than once|duplicate"):
```

Ruff (0.15.15)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)